secio/README.md

```diff
@@ -0,0 +1,160 @@
+# SECIO 1.0.0
+
+> A stream security transport for libp2p. Streams wrapped by SECIO use secure
+> sessions to encrypt all traffic.
+
+## Authors
+
+- [Juan Benet](https://github.com/jbenet) (October, 2015)
+
+## Editors
+
+- [Cole Brown](https://github.com/bigs)
+- [Lars Gierth](https://github.com/lgierth)
+
+## Implementations
+
+- [js-libp2p-secio](https://github.com/libp2p/js-libp2p-secio)
+- [go-secio](https://github.com/libp2p/go-libp2p-secio)
+
+## Table of Contents
+
+- [Algorithm Support](#algorithm-support)
+
+## Algorithm Support
+
+SECIO allows participating peers to support a subset of the following
+algorithms.
+
+### Exhchanges
+
+- P-256
+- P-384
+- P-521
+
+### Ciphers
```

Comment:
Ciphers used for key stretching and for message encryption once SECIO channel is established.

Reply:
Why is a deprecated cipher (Blowfish) supported? From the docs:

>  Blowfish is a legacy cipher and its short block size makes it vulnerable to birthday bound attacks (see https://sweet32.info). It should only be used where compatibility with legacy systems, not security, is the goal.

> Deprecated: any new system should use AES (from crypto/aes, if necessary in an AEAD mode like crypto/cipher.NewGCM) or XChaCha20-Poly1305 (from golang.org/x/crypto/chacha20poly1305).

Reply:
:+1:

Reply:
It would also be worth noting what mode the AES ciphers should use. From the Go implementation, it looks like it's CTR!

Also, more generally, is this an implementation of any well-reviewed specification? If not, is there a reason why? I would imagine there are a number of specs for low-overhead, encrypted channels which may prevent future pitfalls.

Reply:
This is definitely not well-reviewed and will be will be deprecated in favor of TLS quite soon. See: https://github.com/libp2p/go-libp2p-tls/.

(go-ipfs now has experimental support which we'll likely upgrade to default support after a release or two).

Reply:
Gotcha, makes sense! And that answers my follow-up questions as well :)